assert-plus/index.d.ts

```diff
@@ -3,65 +3,505 @@
+ * If the `message` parameter is `undefined`, a default error message is assigned.
+ * 
+ * ```js
+ * const assert = require('assert');
+ * 
+ * assert.ok(true);
+ * // OK
+ * assert.ok(1);
+ * // OK
+ * assert.ok(false);
+ * // throws "AssertionError: false == true"
+ * assert.ok(0);
+ * // throws "AssertionError: 0 == true"
+ * assert.ok(false, 'it\'s false');
+ * // throws "AssertionError: it's false"
+ * ```
+ */
 export function ok(options: any, message ?: string): void;
+
+/**
+ * Tests shallow, coercive equality between the actual and expected parameters
+ * using the equal comparison operator ( `==` ).
+ * 
+ * ```js
+ * const assert = require('assert');
+ * 
+ * assert.equal(1, 1);
+ * // OK, 1 == 1
+ * assert.equal(1, '1');
+ * // OK, 1 == '1'
+ * 
+ * assert.equal(1, 2);
+ * // AssertionError: 1 == 2
+ * assert.equal({a: {b: 1}}, {a: {b: 1}});
+ * //AssertionError: { a: { b: 1 } } == { a: { b: 1 } }
+ * ```
+ * 
+ * If the values are not equal, an `AssertionError` is thrown with
+ * a `message` property set equal to the value of the `message` parameter.
+ * If the `message` parameter is undefined, a default error message is assigned.
+ */
 export function equal(actual: any, expected: any, message ?: string): void;
+
+/**
+ * Tests shallow, coercive inequality with the not equal comparison operator ( `!=` ).
+ * 
+ * ```js
+ * const assert = require('assert');
+ * 
+ * assert.notEqual(1, 2);
+ * // OK
+ * 
+ * assert.notEqual(1, 1);
+ * // AssertionError: 1 != 1
+ * 
+ * assert.notEqual(1, '1');
+ * // AssertionError: 1 != '1'
+ * ```
+ * 
+ * If the values are equal, an `AssertionError` is thrown with
+ * a `message` property set equal to the value of the `message` parameter.
+ * If the `message` parameter is undefined, a default error message is assigned.
+ */
 export function notEqual(actual: any, expected: any, message ?: string): void;
+
+/**
+ * Tests for deep equality between the `actual` and `expected` parameters.
+ * Primitive values are compared with the equal comparison operator ( `==` ).
+ * 
+ * Only enumerable "own" properties are considered.
+ * The `deepEqual()` implementation does not test object prototypes, attached symbols,
+ * or non-enumerable properties. This can lead to some potentially surprising results.
+ * For example, the following example does not throw an `AssertionError` because
+ * the properties on the Error object are non-enumerable:
+ * 
+ * ```js
+ * // WARNING: This does not throw an AssertionError!
+ * assert.deepEqual(Error('a'), Error('b'));
+ * ```
+ * 
+ * "Deep" equality means that the enumerable "own" properties of child objects are evaluated also.
+ * 
+ * If the values are not equal, an `AssertionError` is thrown with a `message` property
+ * set equal to the value of the `message` parameter. If the `message` parameter is undefined,
+ * a default error message is assigned.
+ */
 export function deepEqual(actual: any, expected: any, message ?: string): void;
```

Comment:
Could use `deepEqual<T>(actual: T, expected:T, message?: string)` so that compiler warns for e.g. `deepEqual(myString, myNumber)`.